test/ntttcp_test.py

```diff
@@ -0,0 +1,470 @@
+        # 05:57:38 INFO: Test warmup completed.
+        warmup_completed_pattern = re.compile(r"(\d+) INFO: Test warmup completed.")
+        warmup_completed = warmup_completed_pattern.findall(self.send_stdout)
+        matched_warmup_time = int(warmup_completed[0]) - int(network_start[0])
+        # INFO: test duration    :5 seconds
+        duration_pattern = re.compile(r"INFO: test duration\s+:(\d+)")
+        matched_duration_time = duration_pattern.findall(self.send_stdout)
+        # 05:57:44 INFO: Test cooldown is in progress...
+        cooldown_start_pattern = re.compile(r"(\d+) INFO: Test cooldown is in progress...")
+        cooldown_start = cooldown_start_pattern.findall(self.send_stdout)
+        # 05:57:48 INFO: Test cycle finished.
+        test_cycle_pattern = re.compile(r"(\d+) INFO: Test cycle finished.")
+        test_cycle = test_cycle_pattern.findall(self.send_stdout)
+        matched_cooldown_time = int(test_cycle[0]) - int(cooldown_start[0])
+        if matched_warmup_time and matched_duration_time and matched_cooldown_time:
+            warmup_time = int(matched_warmup_time)
+            duration_time1 = int(matched_duration_time[0])
+            cooldown_time = int(matched_cooldown_time)
+            return warmup_time, duration_time1, cooldown_time
+        else:
+            print("Can not match the running time")
+
+    def is_epoll(self) -> bool:
+        # *** use epoll()
+        epoll_pattern = re.compile(r"\*\*\*\s+use epoll()")
+        if epoll_pattern.findall(self.rec_stdout):
+            return True
+        else:
+            return False
+
+    def get_buffer_size(self) -> int:
+        # receiver socket buffer (bytes):  1470
+        buffer_size_pattern = re.compile(r"receiver socket buffer \(bytes\):\s+(\d+)")
+        buffer_size = buffer_size_pattern.findall(self.rec_stdout)
+        if buffer_size:
+            matched_buffer_size = buffer_size[0]
+            return int(matched_buffer_size)
+        else:
+            print("Can not match the buffer_size")
+
+    def is_output_log_file(self, filename) -> bool:
+        filename_out = open(filename, "r")
+        value_filename_out = filename_out.read()
+        throughput_in_logfile_pattern = re.compile(r"INFO:\s+throughput\s+:(\d+\.\d+)(\w+)")
+        throughput_in_logfile = throughput_in_logfile_pattern.findall(value_filename_out)
+        if subprocess.run(f"ls {filename}", shell=True, check=True) and throughput_in_logfile:
+            print("throughput_in_logfile:", throughput_in_logfile)
+            return True
+        else:
+            return False
+
+    def parseFile(self, filename) -> bool:
+        parser = make_parser()
+        parser.setContentHandler(ContentHandler())
+        parser.parse(filename)
+        print("parser.parse(filename):",parser.parse(filename))
+        if parser.parse(filename):
+            return True
+        else:
+            return False
+
+    def is_output_xml_file(self, filename) -> bool:
```

Comment:
See above is_output_log_file(); I suggest you combine the 3 functions together.

Reply:
done